src/spaces/compartment.jl

```diff
@@ -18,7 +18,10 @@ function random_position(model::ABM{A, <:CompartmentSpace{D}}) where {A,D}
     pos = Tuple(rand(D) .* model.space.dims)
 end
 
-pos2cell(pos::Tuple) = ceil.(Int, pos)
+function pos2cell(pos::Tuple) 
+    any(x-> x <= 0, pos) && throw("Position should be positive")
```

Comment:
This is too costly to be done every single in every call to `nearby_...`, I would suggest to instead state that positions can only be positive in the space documentation string.